lib/twitter/rest/tweets.rb

```diff
@@ -300,6 +296,45 @@ def retweeters_ids(*args)
 
     private
 
+      # Uploads images and videos. Videos require multiple requests and uploads in chunks of 5 Megabytes.
+      # The only supported video format is mp4.
+      #
+      # @see https://dev.twitter.com/rest/public/uploading-media
+      def upload(media) # rubocop:disable MethodLength, AbcSize
+        if !(File.basename(media) =~ /\.mp4$/)
```

Comment:
True, didn't want to add any dependencies or complexity for this here though, iirc.